be/src/vec/sink/vtablet_sink.cpp

```diff
@@ -117,12 +117,17 @@ Status VOlapTableSink::send(RuntimeState* state, vectorized::Block* input_block)
         _partition_to_tablet_map.clear();
     }
     
-    //if pending bytes is more than 500M, wait
+    //if pending bytes is more than 500M, wait at most 1 min
     constexpr size_t MAX_PENDING_BYTES = 500 * 1024 * 1024;
-    if ( get_pending_bytes() > MAX_PENDING_BYTES){
-        while(get_pending_bytes() < MAX_PENDING_BYTES){
-            std::this_thread::sleep_for(std::chrono::microseconds(500));
-        }
+    constexpr int max_retry = 120;
+    int retry = 0;
+    while (get_pending_bytes() > MAX_PENDING_BYTES && retry++ < max_retry) {
+        std::this_thread::sleep_for(std::chrono::microseconds(500));
+    }
+    if (get_pending_bytes() > MAX_PENDING_BYTES){
+        std::stringstream str;
+        str << "Load task " << _load_id << ": Memory exceed limit. ";
```

Comment:
Users may be confused by the error message. To me, it seems that I should use a bigger exec_mem_limit, however, the actual failure is not due to memory exceeds.  Should we really fail the load task here ? If so, may be a message "task stuck" should be reported?

Reply:
more specific error message added
```
str << "Load task " << _load_id
            << ": pending bytes exceed limit (config::table_sink_pending_bytes_limitation):"
            << MAX_PENDING_BYTES;
```
the limitation is not hard coded, it is a config now